Tests/CSharp/TypeCastTests.cs

```diff
@@ -257,33 +542,169 @@ private void Test()
+    {
+        await TestConversionVisualBasicToCSharpAsync(@"
+Enum TestEnum
+    None = 0
+End Enum
+
+Class Class1
+    Sub TestEnumCast(str as String)
+        Dim enm  As TestEnum = str
+    End Sub
+End Class",
+            @"using Microsoft.VisualBasic.CompilerServices; // Install-Package Microsoft.VisualBasic
+
+internal enum TestEnum
+{
+    None = 0
+}
+
+internal partial class Class1
+{
+    public void TestEnumCast(string str)
+    {
+        TestEnum enm = (TestEnum)Conversions.ToInteger(str);
+    }
+}
+");
+    }
+
     [Fact]
     public async Task CastingIntegralTypeToEnumShouldUseExplicitCastAsync()
     {
-        await TestConversionVisualBasicToCSharpAsync(
-            @"
+        await TestConversionVisualBasicToCSharpAsync(@"
 Enum TestEnum
     None = 0
 End Enum
 Enum TestEnum2
     None = 1
 End Enum
 Class Class1
-    Private Sub Test()
-        Dim res = CType(5S, TestEnum)
-        res = CType(5D, TestEnum)
-        res = CType(5L, TestEnum)
-        res = CType(5.4F, TestEnum)
-        res = CType(5.7R, TestEnum)
-        res = CType(5UL, TestEnum)
-        res = CType(5, TestEnum)
-        
-        Dim otherEnum = TestEnum2.None
-        res = CType(otherEnum, TestEnum)
+    Private Sub TestIntegrals(b as Byte, s as Short, i as Integer, l as Long, e as TestEnum2)
+        Dim res = CType(b, TestEnum)
+        res = CType(s, TestEnum)
+        res = CType(i, TestEnum)
+        res = CType(l, TestEnum)
+        res = CType(e, TestEnum)
+    End Sub
+
+    Private Sub TestNullableIntegrals(b as Byte?, s as Short?, i as Integer?, l as Long?, e as TestEnum2?)
+        Dim res = CType(b, TestEnum)
+        res = CType(s, TestEnum)
+        res = CType(i, TestEnum)
+        res = CType(l, TestEnum)
+        res = CType(e, TestEnum)
+    End Sub
+
+    Private Sub TestUnsignedIntegrals(b as SByte, s as UShort, i as UInteger, l as ULong)
+        Dim res = CType(b, TestEnum)
+        res = CType(s, TestEnum)
+        res = CType(i, TestEnum)
+        res = CType(l, TestEnum)
+    End Sub
+
+    Private Sub TestNullableUnsignedIntegrals(b as SByte?, s as UShort?, i as UInteger?, l as ULong?)
+        Dim res = CType(b, TestEnum)
+        res = CType(s, TestEnum)
+        res = CType(i, TestEnum)
+        res = CType(l, TestEnum)
     End Sub
 End Class",
-            @"using Microsoft.VisualBasic.CompilerServices; // Install-Package Microsoft.VisualBasic
+                @"
+internal enum TestEnum
+{
+    None = 0
+}
 
+internal enum TestEnum2
+{
+    None = 1
+}
+
+internal partial class Class1
+{
+    private void TestIntegrals(byte b, short s, int i, long l, TestEnum2 e)
+    {
+        TestEnum res = (TestEnum)b;
+        res = (TestEnum)s;
+        res = (TestEnum)i;
+        res = (TestEnum)l;
+        res = (TestEnum)e;
+    }
+
+    private void TestNullableIntegrals(byte? b, short? s, int? i, long? l, TestEnum2? e)
+    {
+        TestEnum res = (TestEnum)b;
```

Comment:
This one is actually not true but I will fix that in another PR as this one is already big enough.
```csharp
TestEnum res = (TestEnum)b.Value;
```